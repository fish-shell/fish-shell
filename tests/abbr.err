abbr: no such abbreviation 'NOT_AN_ABBR'
abbr: abbreviation cannot have spaces in the key
abbr: no such abbreviation '__abbr6'
abbr: abbreviation cannot have spaces in the key
abbr: option requires an argument -- -r
abbr: Unexpected argument -- __abbr10
abbr: abbreviation '__abbr12' already exists, cannot rename
