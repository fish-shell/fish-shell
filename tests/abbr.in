# Test basic add and list
abbr __abbr1 alpha beta gamma
abbr | grep __abbr1

# Erasing one that doesn't exist should do nothing
abbr --erase NOT_AN_ABBR
abbr | grep __abbr1

# Adding existing one should be idempotent
abbr __abbr1 alpha beta gamma
abbr | grep __abbr1

# Replacing
abbr __abbr1 delta 
abbr | grep __abbr1

# -s and --show tests
abbr -s | grep __abbr1
abbr --show | grep __abbr1

# Test erasing
abbr -e __abbr1
abbr | grep __abbr1

# Ensure we escape special characters on output
abbr '~__abbr2' '$xyz'
abbr | grep __abbr2
abbr -e '~__abbr2'

# Ensure we handle leading dashes in abbreviation names properly
abbr -- '--__abbr3' 'xyz'
abbr | grep __abbr3
abbr -e '--__abbr3'

# Ensure we are not recognizing later "=" as separators
abbr d2 env a=b banana
abbr -l | string match -q d2; or echo "= test failed"
