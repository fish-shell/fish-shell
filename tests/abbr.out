abbr __abbr1 'alpha beta gamma'
abbr __abbr1 'alpha beta gamma'
abbr __abbr1 'alpha beta gamma'
abbr __abbr1 delta
abbr __abbr1 delta
abbr __abbr1 delta
abbr '~__abbr2' '$xyz'
abbr -- --__abbr3 xyz
